src/RevOpts.cc

```diff
@@ -49,226 +26,117 @@ void RevOpts::SetArgs( const SST::Params& params ) {
   }
 }
 
-bool RevOpts::InitPrefetchDepth( const std::vector<std::string>& Depths ) {
-  std::vector<std::string> vstr;
-  for( uint32_t i = 0; i < Depths.size(); i++ ) {
-    std::string s = Depths[i];
+template<typename MAP>
+bool RevOpts::InitPropertyMap( const std::vector<std::string>& Opts, MAP& map ) {
+  for( auto& s : Opts ) {
+    std::vector<std::string> vstr;
+
     splitStr( s, ":", vstr );
     if( vstr.size() != 2 )
       return false;
 
-    uint32_t Core = std::stoul( vstr[0], nullptr, 0 );
-    if( Core > numCores )
+    auto Core = std::stoull( vstr[0], nullptr, 0 );
+    if( Core >= numCores )
       return false;
 
-    std::string::size_type sz          = 0;
-    uint32_t               Depth       = std::stoul( vstr[1], &sz, 0 );
+    // Store as cast integer if target is integer; otherwise store as string
+    auto parse = [&]( auto val ) {
+      if constexpr( std::is_integral_v<decltype( val )> ) {
+        map[Core] = decltype( val )( std::stoull( vstr[1], nullptr, 0 ) );
+      } else {
+        map[Core] = vstr[1];
+      }
+    };
```

Comment:
This lambda represents an anonymous function. The `[&]` makes it capture any expressions it uses in its parent. The `auto val` is an argument which is only used in `decltype( val )` but which is a way of making this lambda a "generic lambda" which is like a template under-the-hood. The `if constexpr` will only execute one of its two branches, for integral and non-integral vector/map elements.